### PR TITLE
Fixes #20006: Store processes of an inventory after storing the inventory

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/20006

### CODE REVIEW

File: webapp/sources/ldap-inventory/inventory-repository/src/main/scala/com/normation/inventory/ldap/core/InventoryMapper.scala [hunk 1]
```diff
@@ -775,6 +773,16 @@ class InventoryMapper(
         win.setOpt(productId, A_WIN_ID, { x: String => x })
         win
     }
+    root
+  }
+
+  def treeFromNode(server:NodeInventory) : LDAPTree = {
```
Comment (fanf) [inline webapp/sources/ldap-inventory/inventory-repository/src/main/scala/com/normation/inventory/ldap/core/InventoryMapper.scala:779]: it needs at least a comment, perhaps even a parameter `includeProcess: Boolean = false`. Else it's sure that one day we will wonder why our nodes are missing process in LDAP

File: webapp/sources/ldap-inventory/inventory-provisioning-core/src/main/scala/com/normation/inventory/ldap/provisioning/DefaultReportSaver.scala [hunk 2]
```diff
@@ -83,11 +83,15 @@ class DefaultReportSaver(
         t3  <- currentTimeMillis
         _   <- InventoryProcessingLogger.timing.trace(s"Saving node: ${t3 - t2} ms")
 
-        d3  <- ZIO.foreach(report.vms) { x =>con.saveTree(mapper.treeFromMachine(x), deleteRemoved = true) }
+        d3  <- con.save(mapper.processesFromNode(report.node), removeMissingAttributes = false)
```
Comment (fanf) [inline webapp/sources/ldap-inventory/inventory-provisioning-core/src/main/scala/com/normation/inventory/ldap/provisioning/DefaultReportSaver.scala:86]: I *think* (not sure) that we could even ignore errors on that one (or at least timeout errors) (with a `catchAll(err => log(err) *> return an empty list of modif for that part)`

**Comment by ncharles** [comment]
PR updated with a new commit

File: webapp/sources/ldap-inventory/inventory-provisioning-core/src/main/scala/com/normation/inventory/ldap/provisioning/DefaultReportSaver.scala (hunk 2 above)
Comment (fanf) [reply webapp/sources/ldap-inventory/inventory-provisioning-core/src/main/scala/com/normation/inventory/ldap/provisioning/DefaultReportSaver.scala]: ha

Comment (fanf) [reply webapp/sources/ldap-inventory/inventory-provisioning-core/src/main/scala/com/normation/inventory/ldap/provisioning/DefaultReportSaver.scala]: oh it's outdated :)

Comment (ncharles) [reply webapp/sources/ldap-inventory/inventory-provisioning-core/src/main/scala/com/normation/inventory/ldap/provisioning/DefaultReportSaver.scala]: i found a solution when i pressed enter :)
i need to check it works

**Review by fanf — APPROVED** [review]

**Comment by fanf** [comment]
OK, squash merging this PR

**Pull request merged**